sphinx_extensions/doxysphinx_theme/__init__.py

```diff
@@ -0,0 +1,43 @@
+# =====================================================================================
+#  C O P Y R I G H T
+# -------------------------------------------------------------------------------------
+#  Copyright (c) 2023 by Robert Bosch GmbH. All rights reserved.
+#
+#  Author(s):
+#  - Aniket Salve, Robert Bosch GmbH
+# =====================================================================================
+
+"""Doxysphinx themes extension.
+
+This extensions add custom javascript files depending on theme.
+Currently only supporting Book and RTD theme.
+"""
+
+import pathlib
+
+from sphinx.application import Sphinx
+from sphinx.config import Config
+
+
+def setup(app: Sphinx):
+    """Setups up the replacer extension."""
+    app.connect("config-inited", doxysphinx_theme_extension)
+    return {"parallel_read_safe": True, "parallel_write_safe": True, "version": "0.1.0"}
+
+
+def doxysphinx_theme_extension(app: Sphinx, config: Config):
+    """Add custom javascript files specific to theme and set theme options.
+
+    :param app: Sphinx app
+    :param config: Sphinx config
+    """
+    current_file_path = pathlib.Path(__file__).parent.resolve()
+
+    config.html_static_path.append(str(current_file_path) + "/_static/")
+
+    if config.html_theme == "sphinx_book_theme":
+        app.add_js_file("js/customize-navbar-book.js")
+
+    elif config.html_theme == "sphinx_rtd_theme":
+        app.add_js_file("js/customize-navbar-rtd.js")
+        config.html_theme_options["collapse_navigation"] = False
```

Comment:
Sorry for being nosey here, but could there be some kind of explicit option to get these files, at the [ROCm documentation](https://github.com/RadeonOpenCompute/rocm-docs-core) our theme is a slightly modified version of the Sphinx Book Theme (custom headers and footers), and I'm certain the js file for that would work for our case, so it would be good if we could enable this fix somehow.

Reply:
We are planning to provide these files as part of an extension which will be part of Doxysphinx. So hopefully you won't need to add anything manually.

Reply:
@Maetveis So you have an own version of the sphinx book theme with a different name, e.g. "sphinx-book-theme-custom" right?  In that case as you said his extension won't do anything for you. 

But you could still use the jsfile as it's in the doxysphinx-theme package directory. 
Maybe a little bit cumbersome to reference though, however here's a rough sketch of the idea (untested!):

```python
#<conf.py>
from pathlib import Path
from importlib.resources import files
# ...
patch_js_path = Path(files("doxysphinx_theme") / "_static/js/customize-navbar-book.js")
html_theme_options = {
  'extra_scripts': [str(patch_js_path)]
}
```

But this might be only working in python 3.11+.

Reply:
I think I'll be able to make something like that work, the only slight issue with it is that I need to "know" implementation details of doxysphinx, I guess that is a price for having "forked" a theme.
I'll try this out and maybe do a PR if I come up with a way to have less coupling. Thank you for the help.